gprofiler/metadata/py_module_version.py

```diff
@@ -0,0 +1,266 @@
+    standard_lib_pattern = re.compile(r"/python(?P<version>\d\.\d\d?)/(?!.*(site|dist)-packages)")
+    py_version = None
+
+    for path in result:
+        match = standard_lib_pattern.search(path)
+        if match is not None:
+            if py_version is None:
+                py_version = _get_python_full_version(process, match.group("version"))
+                if py_version is None:
+                    # No need to continue trying if we failed
+                    return None
+            result[path] = ("standard-library", py_version)  # type: ignore
+
+
+@functools.lru_cache(maxsize=128)
+def _get_dists_files(packages_path: str) -> Dict[str, pkg_resources.Distribution]:
+    """Return a dict of filename: dist for the distributions in packages_path"""
+    path_to_dist = {}
+    for dist in pkg_resources.find_distributions(packages_path):
+        files_iter = _files_from_record(dist) or _files_from_legacy(dist)
+        if files_iter is not None:
+            path_to_dist.update(dict.fromkeys((os.path.join(packages_path, file) for file in files_iter), dist))
+    return path_to_dist
+
+
+_warned_no__normalized_cached = False
+
+
+def _get_packages_versions(result: Dict[str, Optional[Tuple[str, str]]], process: Process):
+    # A little monkey patch to prevent pkg_resources from converting "/proc/{pid}/root/" to "/".
+    # This function resolves symlinks and makes paths absolute for comparison purposes which isn't required
+    # for our usage.
+    if hasattr(pkg_resources, "_normalize_cached"):
+        original__normalize_cache = pkg_resources._normalize_cached
+        pkg_resources._normalize_cached = lambda path: path
+    else:
+        global _warned_no__normalized_cached
+        if not _warned_no__normalized_cached:
+            # Log only once so we don't spam the log
+            logger.warning("Cannot get modules version, pkg_resources has no '_normalize_cached' attribute")
+            _warned_no__normalized_cached = True
+
+        # Not much that we can do, pkg_resources.find_distributions won't work properly
+        return result
+
+    # Make sure to catch any exception. If something goes wrong just don't get the version, we shouldn't
+    # interfere with gProfiler
+    try:
+        for path in result:
+            if not path.startswith("/"):
+                continue
+
+            packages_path = _get_packages_dir(path)
+            if packages_path is None:
+                # This module is (probably) not part of a package
+                continue
+            packages_path = resolve_host_path(process, packages_path)
+            paths_to_dists = _get_dists_files(packages_path)
+            dist_info = paths_to_dists.get(resolve_host_path(process, path))
+            if dist_info is not None:
+                name = _get_package_name(dist_info)
+                if name is not None:
+                    result[path] = (name, dist_info.version)
+    except Exception:
```

Comment:
log it? I know we don't want to spam the log but on the other hand we do want to monitor and get this kind of errors.

Reply:
I want to do it once per namespace, hence https://github.com/Granulate/granulate-utils/pull/17

Reply:
I see but the per namespace can wait to a later stage and we don't want to miss these errors